src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,140 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            head = new Node<>(null, value, null);
+            tail = head;
+            size++;
+        } else {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> getNode = getNode(index);
+        Node<T> getNodePrev = getNode.prev;
+        Node<T> newNode = new Node<>(getNodePrev, value, getNode);
+        if (getNodePrev == null) {
+            head = newNode;
+        } else {
+            getNodePrev.next = newNode;
+        }
+        getNode.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> nodeToReplace = getNode(index);
+        T lastNodeValue = nodeToReplace.item;
+        nodeToReplace.item = value;
+        return lastNodeValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> nodeToRemove = getNode(index);
+        removeNode(nodeToRemove);
+        return nodeToRemove.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = head;
+        while (node != null) {
+            if (node.item == object || node.item != null && node.item.equals(object)) {
+                removeNode(node);
+                return true;
+            }
+            node = node.next;
+        }
         return false;
```

Comment:
Don't forget that you can have `null` as a method parameter in the `remove(T object)` method. Make sure to handle `null` values properly to avoid a NullPointerException.